From 936c9f22adc07a3154708074d4326f8da4a9abaa Mon Sep 17 00:00:00 2001
From: Aaron He <aaron@bluestacks.com>
Date: Fri, 22 Mar 2024 13:39:54 +0800
Subject: [PATCH 133/224] [A13][ROB-12790] Adding protection restrictions for
 home package

1. Do not allow user to disable com.uncube.launcher3
2. Do not allow user to revoke permission of com.uncube.launcher3
---
 .../java/com/android/server/pm/PackageManagerService.java   | 4 +++-
 .../server/pm/permission/PermissionManagerServiceImpl.java  | 6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 2b2d079e2302..f770041f7c6f 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -3777,7 +3777,9 @@ public class PackageManagerService implements PackageSender, TestUtilityService
                 throw new IllegalArgumentException("Invalid new component state: " + newState);
             }
             // If user is trying to disable one of our packages, don't allow it.
-            if (packageName.startsWith("com.bluestacks") || "com.uncube.account".equalsIgnoreCase(packageName)) {
+            if (packageName.startsWith("com.bluestacks")
+                    || "com.uncube.account".equalsIgnoreCase(packageName)
+                    || "com.uncube.launcher3".equalsIgnoreCase(packageName)) {
                 if (newState == COMPONENT_ENABLED_STATE_DISABLED
                     || newState == COMPONENT_ENABLED_STATE_DISABLED_USER
                     || newState == COMPONENT_ENABLED_STATE_DISABLED_UNTIL_USED) {
diff --git a/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java b/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
index 1d4f19ba1999..136d3369f712 100644
--- a/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
+++ b/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
@@ -1492,8 +1492,10 @@ public class PermissionManagerServiceImpl implements PermissionManagerServiceInt
     public void revokeRuntimePermission(String packageName, String permName, int userId,
             String reason) {
 
-        if (BST_REV_PERM && packageName != null &&
-                (packageName.startsWith("com.bluestacks") || "com.uncube.account".equalsIgnoreCase(packageName))) {
+        if (BST_REV_PERM && packageName != null
+                && (packageName.startsWith("com.bluestacks")
+                || "com.uncube.account".equalsIgnoreCase(packageName)
+                || "com.uncube.launcher3".equalsIgnoreCase(packageName))) {
             if (BST_DEBUG) Log.d(TAG, "In revokeRuntimePermission: not revoking permissions for package =" + packageName + ", returning");
             return ;
         }
